web/docs/guides/auth/auth-apple.mdx

```diff
@@ -0,0 +1,215 @@
+		aud: "https://appleid.apple.com",
+		sub: client_id
+	},
+	private_key,
+	"ES256",
+	header_fields=
+	{
+		kid: key_id 
+	}
+)
+puts token
+```
+
+- Edit the `secret_gen.rb` file:
+    - key_file = "Path to the private key you downloaded from Apple"
+        - It should look like this: AuthKey_XXXXXXXXXX.p8
+    - team_id = "Your Team ID"
+        - This is found at the top right of the Apple Developer site (next to your name)
+    - client_id = "The Service ID of the service you created"
+        - This is the "Services ID" you created in the above step "Obtain a Services ID"
+        - If you've lost this ID, you can find it in the Apple Developer Site:
+            - Go to "Certificates, Identifiers & Profiles"
+            - Click "Identifiers" at the left
+            - At the top right drop-down, select "Services IDs"
+            - Find your Identifier in the list (i.e. app.com.acme.roadrunner)
+    - key_id = "The Key ID of the private key"
+        - This can be found in the name of your downloaded secret file (For a file named "AuthKey_XXXXXXXXXX.p8" your key_id is "XXXXXXXXXX")
+        - If you've lost this ID, you can find it in the Apple Developer Site:
+            - Go to "Certificates, Identifiers & Profiles"
+            - Click "Keys" at the left
+            - Click on your newly-created key in the list
+            - Look under "Key ID" to find your key_id
+    - Once you've entered values for all 4 items (key_file, team_id, client_id, key_id) save the `secret_gen.rb` file
+- Generate your `client_secret`
+    - From the command line, run: `ruby secret_gen.rb > client_secret.txt`
+    - Your `client_secret` is now stored in this `client_secret.txt` file
+
+You should now have a `client_id` and `client_secret`
+
+
+### Enter your Github Client ID and Client Secret into your Supabase Project
+
+- Go to your [Supasebase Project Dashboard](https://supabase.io)
+- In the left sidebar, click the "Authentication" icon (near the top)
+- Click "Settings" from the list to go to the "Authentication Settings" page
+- Enter the final (hosted) URL of your app under "Site URL" (this is important)
+- Under "External OAuth Providers" turn "Apple Enabled" to ON
+- Enter your `client_id` and `client_secret` saved in the previous steps
+- Click "Save"
+
+### Add login code to your client app
+
+The JavaScript client code is documented here: [Supabase OAauth Client Code](https://supabase.io/docs/reference/javascript/auth-signin#sign-in-using-third-party-providers)
+
+```js
+const { user, session, error } = await supabase.auth.signIn({
+  provider: 'apple'
+})
+```
+
+Just put this code in a function and create a button or a link or any other UI element to call the function.
+
+```js
+function signInWithGithub() {
+  const { user, session, error } = await supabase.auth.signIn({
+    provider: 'apple'
+  });
+}
+```
+
+To log out:
+
+```js
+function signout() {
+  const { error } = await supabase.auth.signOut();
+}
+```
+
+## Resources
```

Comment:
I think we should add the link to https://medium.com/identity-beyond-borders/how-to-configure-sign-in-with-apple-77c61e336003 since the ruby script was actually borrowed from there.